src/message/MoveMessage.js

```diff
@@ -0,0 +1,216 @@
+  }),
+};
+
+export default function MoveMessage(props: Props): Node {
+  const themeContext = useContext(ThemeContext);
+  const backgroundColor = themeContext.backgroundColor;
+  const cardColor = themeContext.cardColor;
+  const iconName = Platform.OS === 'android' ? 'arrow-left' : 'chevron-left';
+  const auth = useSelector(getAuth);
+  const allStreams = useSelector(getStreams);
+  const isAdmin = useSelector(getOwnUser).is_admin;
+  const messageId = props.route.params.message.id;
+  const currentStreamName = streamNameOfNarrow(props.route.params.messageNarrow);
+  const currentStreamId = allStreams.find(s => s.name === currentStreamName)?.stream_id;
+  const [narrow, setNarrow] = useState(streamNarrow(currentStreamName));
+  const [subject, setSubject] = useState(props.route.params.message.subject);
+  const [propagateMode, setPropagateMode] = useState('change_one');
+  const [streamId, setStreamId] = useState(currentStreamId);
+  const [topicFocus, setTopicFocus] = useState(false);
+
+  const styles = {
+    parent: {
+      backgroundColor: cardColor,
+    },
+    layout: {
+      margin: 10,
+    },
+    title: {
+      fontSize: 18,
+      color: backgroundColor === 'white' ? 'black' : 'white',
+    },
+    topicInput: {
+      height: 50,
+      backgroundColor,
+      ...inputMarginPadding,
+    },
+    viewTitle: {
+      display: 'flex',
+      flexDirection: 'row',
+      justifyContent: 'space-between',
+      alignItems: 'center',
+      height: 50,
+      paddingHorizontal: 10,
+      marginBottom: 20,
+    },
+    textColor: {
+      color: backgroundColor === 'white' ? 'black' : 'white',
+    },
+    picker: { backgroundColor, marginBottom: 20 },
+    submitButton: {
+      marginTop: 10,
+      paddingTop: 15,
+      paddingBottom: 15,
+      marginLeft: 30,
+      marginRight: 30,
+      backgroundColor: BRAND_COLOR,
+      borderRadius: 10,
+      borderWidth: 1,
+    },
+  };
+
+  const handleTopicChange = (topic: string) => {
+    setSubject(topic);
+  };
+
+  const handleTopicFocus = () => {
+    setTopicFocus(true);
+  };
+
+  const setTopicInputValue = (topic: string) => {
+    handleTopicChange(topic);
+    setTopicFocus(false);
+  };
+
+  const handleTopicAutocomplete = (topic: string) => {
+    setTopicInputValue(topic);
+  };
+
+  const updateMessage = () => {
+    if (isAdmin) {
+      api
+        .updateMessage(
+          auth,
+          { subject, stream_id: streamId, propagate_mode: propagateMode },
+          messageId,
+        )
+        .catch(error => {
+          showErrorAlert('Failed to move message', error.message);
+          props.navigation.goBack();
+        });
+    } else {
+      api
+        .updateMessage(auth, { subject, propagate_mode: propagateMode }, messageId)
+        .catch(error => {
+          showErrorAlert('Failed to move message', error.message);
+          props.navigation.goBack();
+        });
+    }
+    props.navigation.goBack();
+    Toast.show('Moved Message');
```

Comment:
Write the confirmation message in sentence case: only the first word capitalized, so like "Moved message".